File: 백준/Silver/1874. 스택 수열/스택 수열.java
import java.io.*;
import java.util.ArrayList;
import java.util.Stack;

public class Main{
    public static void main(String[] args) throws IOException{

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        Stack<Integer> stack = new Stack<>();

        int n = Integer.parseInt(br.readLine());
        int element = 1;
        ArrayList<String> answer = new ArrayList<>();

        for (int i=0; i<n; i++) {
            int num = Integer.parseInt(br.readLine());

            while (element <= num) {
                stack.push(element++);
                answer.add("+\n");
            }

            if (stack.peek() == num) {
                stack.pop();
                answer.add("-\n");
            }
        }

        if (stack.isEmpty()) {
            for (String word : answer)
                bw.write(word);
        } else {
            bw.write("NO");
        }

        bw.flush();
    }
}

File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys
from collections import deque

if __name__ == '__main__':
    n = int(sys.stdin.readline())
    element = 1
    stack = deque()
    output = []
    for _ in range(n):
        num = int(sys.stdin.readline())

        while element <= num:
            stack.append(element)
            output.append('+')
            element += 1

        if stack[-1] == num:
            stack.pop()
            output.append('-')

    if stack:
        print('NO')
    else:
        print('\n'.join(output))